cli/internal/auth/team.go

```diff
@@ -2,23 +2,46 @@ package auth
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/http"
 	"os"
+	"strings"
 
 	cqapi "github.com/cloudquery/cloudquery-api-go"
 	"github.com/cloudquery/cloudquery-api-go/auth"
 	"github.com/cloudquery/cloudquery-api-go/config"
 	"github.com/cloudquery/cloudquery/cli/internal/api"
+	teamapi "github.com/cloudquery/cloudquery/cli/internal/team"
 )
 
+func getAvailableUserTeams(ctx context.Context, token auth.Token) []string {
+	cl, err := teamapi.NewClient(token.Value)
+	if err != nil {
+		return nil
+	}
+	teams, err := cl.ListAllTeams(ctx)
+	if err != nil {
+		return nil
+	}
+	return teams
+}
+
 // GetTeamForToken returns the team for the given token
 // If the token is a bearer token, we need to get the team from the configuration.
 // For api keys the team name is not required as the key is bound to a team name already.
 func GetTeamForToken(ctx context.Context, token auth.Token) (string, error) {
 	switch token.Type {
 	case auth.BearerToken:
 		team, err := config.GetValue("team")
+		if errors.Is(err, os.ErrNotExist) {
```

Comment:
fyi: This is done slightly differently in [GetTeamForAnyToken](https://github.com/cloudquery/cloudquery/blob/577fc65b374c9234fc0b79a2fe5507bf8492ce7b/cli/internal/publish/addons.go)

Reply:
Merged it into the updated code in https://github.com/cloudquery/cloudquery/pull/18225/commits/e68603ee5ee7c1da5783d18239ea5f184b3b8566